### PR TITLE
[CK SDPA] Add CK backend support for varlen attention (#178322) (#178322)

### PR DESCRIPTION
**Pull request by haoyuz**
Summary:
Port GitHub PR https://github.com/pytorch/pytorch/issues/172246 to internal. This adds composable_kernel (CK) backend
support for scaled-dot-product-attention varlen path on ROCm:

- Fix philox seed/offset access in varlen backward to guard on is_dropout
- Fix softmax_lse shape and stride for CK varlen forward path
- Add `is_ck_sdpa_available()` API to `torch.backends.cuda`
- Update varlen_attn fake to use correct logsumexp shape per backend
- Add CK backend parametrization to varlen attention tests

Co-authored-by: Chinmay Kuchinad ChinmayDattanand.Kuchinad@amd.com
Co-authored-by: Jeff Daily jeff.daily@amd.com

Test Plan: buck test on relevant test targets

Reviewed By: yinbinm

Differential Revision: D97799113

Pulled By: haoyuz

### CODE REVIEW

**Comment by haoyuz** [comment]
> @haoyuz linter error is real.

@jeffdaily Fix the linter errors. Thanks!



> @haoyuz the description mentions "Add new fmha_fwd_args fields (block_scale pointers, descale/batch strides, block_scale sizes) in CK forward kernels" but I don't think I see those changes in this diff. Might be worth updating the description to match?

@chinmaydk99 Update the comment, PTAL. Thanks!

**Review by jeffdaily — APPROVED** [review]
